### PR TITLE
B.4.6 add complain reason

### LINKED ISSUE #163: B.4.5 Delete Offensive Comments

### LINKED ISSUE #164: B.4.6 Add Complaint Reason to list of Complaint Reasons

### LINKED ISSUE #165: B.4.7 View List of Complaints

### LINKED ISSUE #166: B.4.8 View a Complaint

### LINKED ISSUE #167: B.4.9 Mark Complaint as Resolved

### PR DESCRIPTION
**Pull request by kengthong**
### Changelog:
1. Created complaint and complaintReason entity
2. Endpoints added:
- Add complaint reason
- Get complaint reasons
- Get all complaints by filterr
- Add a comment complaint
- Resolve a complaint

## Description:
_Eg; Fixed router bug_
## Checklist:
- [x] Merged latest develop
- [x] PR title makes sense

### CODE REVIEW

File: src/controllers/comment.controller.ts [hunk 1]
```diff
@@ -1,81 +1,41 @@
...
+  RESPONSE_ERROR,
+  SOCIAL_ERRORS,
+} from '../constants/errors';
 import { COMMENT_RESPONSE } from '../constants/successMessages';
 import CommentService from '../services/comment.service';
 import apiResponse from '../utilities/apiResponse';
 
 export class CommentController {
-// ======================================== COMMENTS FOR LESSON ========================================
-    public static async createLessonComment(req, res) {
```
Comment (kengthong) [inline src/controllers/comment.controller.ts:10]: Method was not deleted, just shifted downwards within the same file

File: src/controllers/comment.controller.ts [hunk 2]
```diff
@@ -1,81 +1,41 @@
...
-              message: e.message,
-            });
-          }
-          return apiResponse.error(res, httpStatusCodes.INTERNAL_SERVER_ERROR, {
-            message: RESPONSE_ERROR.RES_ERROR,
-          });
-        }
-      }
-    
-      public static async getLessonComments(req, res) {
```
Comment (kengthong) [inline src/controllers/comment.controller.ts:42]: This was not deleted too

File: src/controllers/comment.controller.ts [hunk 3]
```diff
@@ -1,81 +1,41 @@
...
-        } catch (e) {
-          logger.error('[commentController.getLessonComments]:' + e.message);
-          return apiResponse.error(res, httpStatusCodes.INTERNAL_SERVER_ERROR, {
-            message: RESPONSE_ERROR.RES_ERROR,
-          });
-        }
-      }
-
-// ======================================== COMMENTS FOR POST ========================================
-public static async createPostComment(req, res) {
```
Comment (kengthong) [inline src/controllers/comment.controller.ts:60]: This was not deleted too

File: src/controllers/comment.controller.ts [hunk 4]
```diff
@@ -87,30 +47,23 @@ public static async createPostComment(req, res) {
     }
   }
 
-public static async editPostComment(req, res) {
```
Comment (kengthong) [inline src/controllers/comment.controller.ts:90]: Method refactored from editPostComment to editComment (comments are available for both Posts and Lessons)

File: src/controllers/comment.controller.ts [hunk 5]
```diff
@@ -122,22 +75,31 @@ public static async editPostComment(req, res) {
     }
   }
 
-  public static async deletePostComment(req, res) {
```
Comment (kengthong) [inline src/controllers/comment.controller.ts:125]: deletePostComment was refactored into deleteComment

File: src/routes/admin.route.ts [hunk 6]
```diff
@@ -36,64 +36,13 @@ const router = express.Router();
...
 const schemaValidator = require('express-joi-validation').createValidator({});
 
+// ======================================== ADMIN AUTH ========================================
 router.post(
   '/login',
   schemaValidator.body(auth.login),
   Utility.asyncHandler(AdminController.login)
 );
 
-router.get(
```
Comment (kengthong) [inline src/routes/admin.route.ts:45]: Methods in admin.route were not deleted, just reorganized the methods

File: src/services/admin.service.ts [hunk 7]
```diff
@@ -11,68 +11,7 @@ import { Admin } from '../models/Admin';
 import { User } from '../models/User';
 import EmailService from './email.service';
 export default class AdminService {
-  public static async deactivateAdmin(
```
Comment (kengthong) [inline src/services/admin.service.ts:14]: Methods in admin service were not deleted, just reorganized the methods

File: src/routes/comment.route.ts [hunk 8]
```diff
@@ -1,64 +1,57 @@
...
-    Utility.asyncHandler(CommentController.editPostComment)
-  );
-
-//delete comment
-router.delete(
-    '/post/:commentId',
-    passport.authenticate('isAuthenticated', { session: false }),
-    schemaValidator.params(comment.commentIdP),
-    Utility.asyncHandler(CommentController.deletePostComment)
-  );
```
Comment (kengthong) [inline src/routes/comment.route.ts:58]: Reeorganised methods to be placed at the top

File: src/services/comment.service.ts [hunk 9]
```diff
@@ -44,59 +82,21 @@ public static async createLessonComment(
...
-    if (comment.accountId !== accountId)
-      throw new Error(
-        httpStatusCodes.getStatusText(httpStatusCodes.UNAUTHORIZED)
-      );
-    await Comment.destroy({
-      where: {
-        commentId,
-      },
-    });
-  }
```
Comment (kengthong) [inline src/services/comment.service.ts:99]: Methods were not deleted, just refactored and reorganized to the top

**Review by ianleck — APPROVED** [review]

**Pull request merged**